src/libmongoc/src/mongoc/mongoc-server-description.c

```diff
@@ -787,7 +787,12 @@ mongoc_server_description_handle_hello (mongoc_server_description_t *sd,
 
    EXIT;
 
-failure:
+typefailure:
+   bson_set_error (&sd->error,
+                   MONGOC_ERROR_STREAM,
+                   MONGOC_ERROR_STREAM_INVALID_TYPE,
+                   "invalid type sent to hello");
```

Comment:
```suggestion
   bson_set_error (&sd->error,
                   MONGOC_ERROR_STREAM,
                   MONGOC_ERROR_STREAM_INVALID_TYPE,
                   "unexpected type %s for field %s in hello response",
                   _mongoc_bson_type_to_str (bson_iter_type (&iter)),
                   bson_iter_key (&iter));
```

Suggest adding the expected type and field to help diagnose the cause of the error.